rand_core/src/le.rs

```diff
@@ -16,16 +16,16 @@ use core::convert::TryInto;
 /// Reads unsigned 32 bit integers from `src` into `dst`.
 #[inline]
 pub fn read_u32_into(src: &[u8], dst: &mut [u32]) {
-    for i in 0..dst.len() {
-        dst[i] = u32::from_le_bytes((&src[i..i + 4]).try_into().unwrap());
+    for (out, chunk) in dst.iter_mut().zip(src.chunks(4)) {
```

Comment:
Should we start by asserting that `src` is long enough? Potentially this avoids the need to test each chunk.

Reply:
Ideally this part should use `chunks_exact` and (debug) assert should check that remainder is empty.

Reply:
I implemented both, minus checking the remainder.